pkg/server/ca/upstream_client_test.go

```diff
@@ -183,26 +176,29 @@ func TestUpstreamClientPublishJWTKey_HandlesBundleUpdates(t *testing.T) {
 }
 
 func TestUpstreamClientPublishJWTKey_NotImplemented(t *testing.T) {
-	client, _, _, uaDone := setUpUpstreamClientTest(t, fakeupstreamauthority.Config{
+	client, _, _ := setUpUpstreamClientTest(t, fakeupstreamauthority.Config{
 		TrustDomain:           "example.org",
 		DisallowPublishJWTKey: true,
 	})
-	defer client.Close()
-	defer uaDone()
 
 	jwtKeys, err := client.PublishJWTKey(context.Background(), &common.PublicKey{Kid: "KEY"})
 	spiretest.RequireGRPCStatus(t, err, codes.Unimplemented, "disallowed")
 	require.Nil(t, jwtKeys)
 }
 
-func setUpUpstreamClientTest(t *testing.T, config fakeupstreamauthority.Config) (*ca.UpstreamClient, *fakeBundleUpdater, *fakeupstreamauthority.UpstreamAuthority, func()) {
-	plugin, upstreamAuthority, done := fakeupstreamauthority.Load(t, config)
+func setUpUpstreamClientTest(t *testing.T, config fakeupstreamauthority.Config) (*ca.UpstreamClient, *fakeBundleUpdater, *fakeupstreamauthority.UpstreamAuthority) {
+	plugin, upstreamAuthority := fakeupstreamauthority.Load(t, config)
 	updater := newFakeBundleUpdater()
 
-	return ca.NewUpstreamClient(ca.UpstreamClientConfig{
+	client := ca.NewUpstreamClient(ca.UpstreamClientConfig{
 		UpstreamAuthority: plugin,
 		BundleUpdater:     updater,
-	}), updater, upstreamAuthority, done
+	})
+	t.Cleanup(func() {
+		assert.NoError(t, client.Close())
```

Comment:
Does this actually work? I think cleanup runs after the test has finished, so I'm not sure the assert.NoError will have the expected result

Reply:
Good question. I'll verify.

Reply:
It does fail the test:
```
package ohno

import "testing"

func TestCleanup(t *testing.T) {
	t.Cleanup(func() {
		t.Error("oh no")
	})
}
```

```
:!go test
--- FAIL: TestCleanup (0.00s)
    ohno_test.go:7: oh no
FAIL
exit status 1
```